rust/protocol/src/sealed_sender.rs

```diff
@@ -757,37 +757,56 @@ pub async fn sealed_sender_encrypt_from_usmc<R: Rng + CryptoRng>(
     Ok(serialized)
 }
 
+/// This is a single-key multi-recipient KEM, defined in Manuel Barbosa's
+/// ["Randomness Reuse: Extensions and Improvements"]. It uses the "Generic Construction" in `4.1`
+/// of that paper, instantiated with ElGamal encryption.
+///
+/// ["Randomness Reuse: Extensions and Improvements"]: https://haslab.uminho.pt/mbb/files/reuse.pdf
+///
+/// ### Overview
+/// 1. generate a random [sealed_sender_v2::DerivedKeys].
+/// 2. encrypt the keypair using a shared secret derived from the sender's private key and
+///    recipient's public key (this secret never changes for that pair of identity keys) with
+///    [sealed_sender_v2::apply_agreement_xor].
+/// 3. sign the encrypted keypair using the sender's private key with
+///    [sealed_sender_v2::compute_authentication_tag].
+/// 4. use the keypair from (1) to symmetrically encrypt the underlying
+///    [UnidentifiedSenderMessageContent].
+/// 5. As described in ["Randomness Reuse: Extensions and Improvements"], the randomness from (1) is
+///    reused for all of the intended recipients of the message.
+///
+/// As per (5), the method described in `4.1` of ["Randomness Reuse: Extensions and Improvements"]
+/// enables reusing randomness generated in (1) in order to reduce computation time for clients
+/// sending the same message to multiple recipients without compromising the message security. The
+/// mechanism is summarized in pseudocode below.
+///
+/// ### Pseudocode
+///```notrust
```

Comment:
Nitpick: The rustdoc book suggests `text`, which I like a little better because I initially read this as "no trust" rather than "not rust"!

Reply:
Oh yes!! I thought that was kind of funny but it's a good idea to be clear about our terminology when we get the chance to be in these matters :)